### DISCUSSION TITLE
Facing problem in add product via listing items API of SP-API

### DISCUSSION THREAD

**Discussion by kishanK111** [amzn/selling-partner-api-models#3460] [General]
I am trying to add product with listing items API of SP-API. I have not any option to set productType unless 'PRODUCT' with requirements set to 'LISTING_OFFER_ONLY'. After API call, Product added successfully to inventory list of Amazon seller account but status of that product is incomplete. In API call of product details status is showing ["DISCOVERABLE"]. It should be ["BUYABLE",  "DISCOVERABLE"]. After save that product in edit form of amazon seller account(I am not changing any details of product), status of product is active. In my observation, problem is in data type of list_price of attributes of body in API call.
I am call api with below parameters in NodeJs.

````
let feed_creation_infos = await sellingPartner(user.amazonSeller.refreshToken).callAPI({
        operation: 'putListingsItem',
        query: { marketplaceIds: ['A21TJRUUN4KGV'] },
        path: {
          sku: params.sku,
          sellerId: user.amazonSeller.sellerId,
        },
        body: {
          productType: 'PRODUCT',
          attributes: {
            merchant_suggested_asin: [
              {
                value: params.asin,
                marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
            condition_type: [
              {
                value: 'new_new',
                marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
            external_product_information: [
              {
                entity: 'HSN Code',
                value: params.hsnCode,
                marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
            country_of_origin: [
              {
                value: 'IN',
                marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
            purchasable_offer: [
              {
                currency: 'INR',
                start_at: {
                  value: new Date(),
                },
                end_at: {
                  value: null,
                },
                our_price: [
                  {
                    schedule: [
                      {
                        value_with_tax: params.sellingPrice,
                      },
                    ],
                  },
                ],
                list_price: [
                  {
                    schedule: [
                      {
                        value_with_tax: params.mrp,
                      },
                    ],
                  },
                ],
               marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
            list_price: [
              {
                currency: 'INR',
                value_with_tax: params.mrp.toString(),
                marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
            fulfillment_availability: [
              {
                fulfillment_channel_code: 'DEFAULT',
                quantity: params.count,
                marketplace_id: 'A21TJRUUN4KGV',
              },
            ],
          },
          requirements: 'LISTING_OFFER_ONLY',
        },
      });
```
`

**Comment by vaidas-lungis** [comment]
```
value_with_tax: params.mrp.toString(),
```
Not a JS dev, but maybe it should be number (float) instead of string?
but its for the `list_price` attribute, and not in `purchasable_offer`. 
interesting..

After the submission did you check errors of the listing? Either notification via the Notification API or trying to get listing via the SP-API ?

**Comment by sergiounabat** [comment]
Hello, same problem here. Did you manage to fix it?

**Comment by rugved1991** [comment]
Hi,

Please check the Product Definition schema and verify the datatype of the attribute. The attribute "value_with_tax" or "value" for list_price or purchaseable_offer should be a number and not a string. 
You should be able to receive the error message in the Listings API response regarding this issue. If you do see an Accepted response, please check the "issues" section of the listing by calling the [getListingsItem API](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#getlistingsitem).

Best,
Rugved
Solutions Architect, SP-API

**Reply by sergiounabat** [reply]
Hi, 

In my case I'm doing it in Java, the data type is double correctly, and it doesn't return any issue in the getListingsItem API call.
